lib/net.js

```diff
@@ -1022,6 +1025,7 @@ function afterConnect(status, handle, req, readable, writable) {
 
   assert.ok(self._connecting);
   self._connecting = false;
+  self._sockname = null;
```

Comment:
Yes, it probably is already be `null` here, but to me this also is the most logical place to be invalidating the cached value.